packages/preset-uno/src/rules/static.ts

```diff
@@ -84,14 +84,100 @@ export const textDecorations: Rule[] = [
   ['no-underline', { 'text-decoration': 'none' }],
 ]
 
+export const textDecorationStyles: Rule[] = [
+  ['underline-solid', {
+    '-webkit-text-decoration-style': 'solid',
+    'text-decoration-style': 'solid',
+  }],
+  ['underline-double', {
+    '-webkit-text-decoration-style': 'double',
+    'text-decoration-style': 'double',
+  }],
+  ['underline-dotted', {
+    '-webkit-text-decoration-style': 'dotted',
+    'text-decoration-style': 'dotted',
+  }],
+  ['underline-dashed', {
+    '-webkit-text-decoration-style': 'dashed',
+    'text-decoration-style': 'dashed',
+  }],
+]
+
 export const textAligns: Rule[] = [
   ['text-center', { 'text-align': 'center' }],
   ['text-left', { 'text-align': 'left' }],
   ['text-right', { 'text-align': 'right' }],
   ['text-justify', { 'text-align': 'justify' }],
 ]
 
-export const listStyleType: Rule[] = [
+export const fontStyles: Rule[] = [
+  ['italic', { 'font-style': 'italic' }],
+  ['not-italic', { 'font-style': 'normal' }],
+]
+
+export const fontSmoothings: Rule[] = [
+  ['antialiased', {
+    '-webkit-font-smoothing': 'antialiased',
+    '-moz-osx-font-smoothing': 'grayscale',
+  }],
+  ['subpixel-antialiased', {
+    '-webkit-font-smoothing': 'auto',
+    '-moz-osx-font-smoothing': 'auto',
+  }],
+]
+
+export const hyphens: Rule[] = [
+  ['hyphens-none', {
+    '-webkit-hyphens': 'none',
+    '-ms-hyphens': 'none',
+    'hyphens': 'none',
+  }],
+  ['hyphens-manual', {
+    '-webkit-hyphens': 'manual',
+    '-ms-hyphens': 'manual',
+    'hyphens': 'manual',
+  }],
+  ['hyphens-auto', {
+    '-webkit-hyphens': 'auto',
+    '-ms-hyphens': 'auto',
+    'hyphens': 'auto',
+  }],
+]
+
+export const writingModes: Rule[] = [
+  ['write-normal', {
+    '-webkit-writing-mode': 'horizontal-tb',
+    '-ms-writing-mode': 'lr-tb',
+    'writing-mode': 'horizontal-tb',
+  }],
+  ['write-vertical-right', {
+    '-webkit-writing-mode': 'vertical-rl',
+    '-ms-writing-mode': 'tb-rl',
+    'writing-mode': 'vertical-rl',
```

Comment:
Most of them are actually suppress 95% coverage on caniuse, I think we could omit those prefixed ones.